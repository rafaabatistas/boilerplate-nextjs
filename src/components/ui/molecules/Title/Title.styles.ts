import styled from 'styled-components';

export const Wrapper = styled.div`
  width: auto;
  height: auto;
  display: flex;
  align-items: center;
  margin-bottom: 20px;
`;

export const Icon = styled.img`
  width: 50px;
  height: 50px;
  margin-right: 15px;
`;

export const Title = styled.h1`
  font-size: 2.8rem;
  color: white;
  font-weight: bold;
`;
