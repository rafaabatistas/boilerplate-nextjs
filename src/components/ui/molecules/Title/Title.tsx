import * as S from './Title.styles';

type TitleProps = {
  icon: string;
  alt: string;
  title: string;
};

export const Title = ({ icon, alt, title }: TitleProps) => {
  return (
    <S.Wrapper>
      <S.Icon src={icon} alt={alt} />
      <S.Title>{title}</S.Title>
    </S.Wrapper>
  );
};
