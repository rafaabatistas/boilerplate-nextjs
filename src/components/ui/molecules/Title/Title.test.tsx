import { render } from '@testing-library/react';

import { Title } from './Title';

describe('<Title />', () => {
  it('Deve renderizar o componente <Title />', () => {
    //Arrange
    const props = {
      icon: 'assets/img/icon-react.png',
      alt: '√çcone com logotipo do nextJS e ao lado escrito Boilerplate NextJS',
      title: 'Boilerplate NextJS'
    };
    const { container } = render(<Title {...props} />);
    //Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
