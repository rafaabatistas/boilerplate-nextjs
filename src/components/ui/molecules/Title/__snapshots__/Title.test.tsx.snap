// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Deve renderizar o componente <Title /> 1`] = `
.c0 {
  width: auto;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c1 {
  width: 50px;
  height: 50px;
  margin-right: 15px;
}

.c2 {
  font-size: 2.8rem;
  color: white;
  font-weight: bold;
}

<div
  class="c0"
>
  <img
    alt="Ícone com logotipo do nextJS e ao lado escrito Boilerplate NextJS"
    class="c1"
    src="assets/img/icon-react.png"
  />
  <h1
    class="c2"
  >
    Boilerplate NextJS
  </h1>
</div>
`;
